perun/connector/adapters/LdapAdapter.py

```diff
@@ -325,52 +403,112 @@ def get_resource_capabilities_by_facility(
+    ) -> List[str]:
         if facility is None:
             return []
         facility_id = AdapterInterface.get_object_id(facility)
         facility_capabilities = self.connector.search_for_entity(
             self._ldap_base,
-            '(&(objectClass=perunFacility)(entityID=' +
-            str(facility_id) + '))',
-            ['capabilities']
+            "(&(objectClass=perunFacility)(entityID=" + str(facility_id) + "))",
+            ["capabilities"],
         )
 
         if not facility_capabilities:
             return []
 
-        return facility_capabilities['capabilities']
+        return facility_capabilities["capabilities"]
 
     def get_facility_capabilities_by_rp_id(self, rp_identifier: str) -> List[str]:
         facility = self.get_facility_by_rp_identifier(rp_identifier)
         return self.get_facility_capabilities_by_facility(facility)
 
-    def _create_internal_representation_group(
-            self, group: dict[str, str]
-    ) -> Group:
+    def _create_internal_representation_group(self, group: dict[str, str]) -> Group:
         return Group(
-            int(group['perunGroupId']),
-            self.get_vo(vo_id=int(group['perunVoId'])),
-            group['uuid'],
-            group['cn'][0],
-            group['perunUniqueGroupName'],
-            group['description'][0] or ''
+            int(group["perunGroupId"]),
+            self.get_vo(vo_id=int(group["perunVoId"])),
+            group["uuid"],
+            group["cn"][0],
+            group["perunUniqueGroupName"],
+            group["description"][0] or "",
         )
+
+    def _get_attributes(self, perun_attr_names, filters):
+        ldap_attr_names = self._get_ldap_attr_names(perun_attr_names)
+
+        ldap_attrs = self.connector.search_for_entities(
+            self._ldap_base, filters, ldap_attr_names
+        )
+        return self._create_attr_name_value_dict(ldap_attrs, perun_attr_names)
+
+    def _create_attr_name_value_dict(self, ldap_attrs, perun_attr_names):
+        attr_values_dict = {}
+        if not ldap_attrs:
+            return attr_values_dict
+
+        for (
+            internal_name,
+            internal_attr_cfg,
+        ) in self._attribute_utils.get_specific_attrs_config_dict(
+            perun_attr_names
+        ).items():
+            print(ldap_attrs)
+            attr_values_dict[internal_name] = self._resolve_attr_value(
+                ldap_attrs[0], internal_attr_cfg
+            )
+        return attr_values_dict
+
+    def _resolve_attr_value(self, ldap_attrs, internal_attr_cfg):
+        if not ldap_attrs[internal_attr_cfg[AttributeUtils.LDAP]]:
+            if internal_attr_cfg[AttributeUtils.TYPE] == AttributeUtils.PERUN_BOOL:
+                return False
+            if internal_attr_cfg[AttributeUtils.TYPE] == AttributeUtils.PERUN_LIST:
+                return []
+            if internal_attr_cfg[AttributeUtils.TYPE] == AttributeUtils.PERUN_DICT:
+                return {}

```

Comment:
```suggestion
            if internal_attr_cfg[AttributeUtils.TYPE] == AttributeUtils.PERUN_DICT:
                return {}
            if internal_attr_cfg[AttributeUtils.TYPE] == AttributeUtils.PERUN_STRING:
                return ""
```
empty value of string should be an empty string, not `None`